pkgs/build-support/rust/fetch-cargo-vendor-util.py

```diff
@@ -21,8 +22,17 @@ def load_toml(path: Path) -> dict[str, Any]:
 
 
 def download_file_with_checksum(url: str, destination_path: Path) -> str:
+    retries = Retry(
+        total=5,
+        backoff_factor=0.5,
+        status_forcelist=[500, 502, 503, 504]
+    )
+    session = requests.Session()
+    session.mount('http://', HTTPAdapter(max_retries=retries))
+    session.mount('https://', HTTPAdapter(max_retries=retries))
+
     sha256_hash = hashlib.sha256()
-    with requests.get(url, stream=True) as response:
+    with session.get(url, stream=True) as response:
```

Comment:
I liked the solution from the other PR, was pretty straightforward

But I guess this is more resilient